hangman.rb

```diff
@@ -0,0 +1,192 @@
+require "colorize"
+
+class Hangman
+
+  def initialize
+
+    @answer_bank = ["cupcake", "spaghetti", "cookie", "hamburger", "pizza", "sandwich", "crackers", "salad", "cheese", "burrito"] # possible hangman words
+    @guesses = 4 # number of allowed incorrect guesses
+    @num_correct = 0 # counter to compare to length of answer
+    @current_answer = @answer_bank[rand(@answer_bank.length)] # string (of answer)
+    @current_answer_array = @current_answer.split(//) # array (of answer)
+    @fillable_answer = [] # array (of blanks to be filled in by answer)
+    @alphabet = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"] # acceptable input
+    @already_guessed = [] # array (to store already guessed letters)
+
+    @current_answer_array.length.times do |x|
+      @fillable_answer[x] = " _ "
+    end
+
+    draw_art
+    print_word
+    play
+  end # initialize
+
+  # DRAWS THE ASCII ART
+  def draw_art
+    # parts of our hangman to be drawn
+    head = "(_)".colorize(:red)
+    arms = "\\|/".colorize(:light_black)
+    torso = "|".colorize(:purple)
+    legs = "/ \\".colorize(:magenta)
+
+    art = <<-END
+|     _________
+|     |/      |
+|     |
+|     |
+|     |
+|     |
+|     |
+|_____|___________
+
+    END
+
+    if @guesses == 3
+      art = <<-END
+|     _________
+|     |/      |
+|     |      #{head}   - "Is it getting hot in here?"
+|     |
+|     |
+|     |
+|     |
+|_____|___________
+
+    END
+
+    elsif @guesses == 2
+      art = <<-END
+|     _________
+|     |/      |
+|     |      #{head}   - "Should I wave my life goodbye?"
+|     |      #{arms}
+|     |
+|     |
+|     |
+|_____|___________
+
+    END
+
+    elsif @guesses == 1
+      art = <<-END
+|     _________
+|     |/      |
+|     |      #{head}   - "I can't feel my legs."
+|     |      #{arms}
+|     |       #{torso}
+|     |
+|     |
+|_____|___________
+
+    END
+
+    elsif @guesses == 0
+      art = <<-END
+|     _________
+|     |/      |
+|     |      #{head}   - "You need to learn
+|     |      #{arms}             how to spell,
+|     |       #{torso}                you killed me!"
+|     |      #{legs}
+|     |
+|_____|___________
+
+    END
+    end
+
+    puts art
+  end #draw_art
+
+  # PRINTS TEXT INFO
+  def print_word
+    print "WORD: "
+    puts @fillable_answer.join
+    puts "" # inserts blank lines
+    puts "Letters Guessed: #{@already_guessed.join(" ")}"
+    puts "Incorrect Guesses Left: #{@guesses}"
+    puts ""
+  end # print_word
+
+  # PLAYING THE GAME
+  def play
+    while @guesses > 0
+
+      print "Type a letter: "
+      letter = gets.chomp.downcase
+      wrong = true
+
+      # triggers if user entered a single character guess
+      if letter.length == 1
+
+        # handles inappropriate user input; only allowing the alphabet for single-character input
+        until @alphabet.include?(letter.upcase)
```

Comment:
Good job figuring out a way of checking for valid characters. Since we're only allowing the letters A-Z, there's an easy way to check without having to define all the characters.

`String` implements `Comparable`, which adds the `between?` method that can be used in this case:
```
until letter.upcase.between?('A', 'Z')
...
end
```